sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py

```diff
@@ -179,12 +181,26 @@ def __init__(self, endpoint, credential, **kwargs):
 
         self._endpoint = endpoint
 
-        self._client = _Client(
-            endpoint=endpoint,
-            sdk_moniker=SDK_MONIKER,
-            authentication_policy=MetricsAdvisorKeyCredentialPolicy(credential),
-            **kwargs
-        )
+        if isinstance(credential, MetricsAdvisorKeyCredential):
+            self._client = _Client(
+                endpoint=endpoint,
+                sdk_moniker=SDK_MONIKER,
+                authentication_policy=MetricsAdvisorKeyCredentialPolicy(credential),
+                **kwargs
+            )
+        else:
+            scope = "https://cognitiveservices.azure.com/.default"
```

Comment:
We shouldn't hardcode the scopes, if a user is using a different cloud the [value will be slightly different](https://github.com/Azure/azure-sdk-for-net/issues/17192) so we need to allow for a custom value to be passed. 

The generated code allows for this, though, so I think we just need to regenerate with the `--add-credentials` flag which will give us [modifiable credential scopes](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_generated/_configuration.py#L46) and default to the `BearerTokenCredentialPolicy` so we don't have to add it.

Reply:
There is an open PR for the swagger https://github.com/Azure/azure-rest-api-specs/pull/10862. Let's regenerate the code when it is settled.